ssl/ssl_lib.c

```diff
@@ -2023,11 +2052,15 @@ long SSL_ctrl(SSL *s, int cmd, long larg, void *parg)
         else
             return 0;
     case SSL_CTRL_SET_MIN_PROTO_VERSION:
-        return ssl_set_version_bound(s->ctx->method->version, (int)larg,
-                                     &s->min_proto_version);
+        return ssl_check_allowed_versions(s->min_proto_version,
+                                          s->max_proto_version)
```

Comment:
Whoops, worrying too hard about efficient editor use and not enough about reading the logic.  Thanks for spotting it.